app/models/piece.rb

```diff
@@ -3,6 +3,20 @@ class Piece < ActiveRecord::Base
 
   enum color: [:white, :black]
 
+  def move_to!(new_x, new_y)
+    piece = Game.find(game_id).pieces.find_by(x_coordinate: new_x, y_coordinate: new_y)
+    if !piece.nil?
+      if piece.color == color
+        'Error'
+      elsif piece.color != color
+        Game.find(game_id).pieces.find_by(x_coordinate: new_x, y_coordinate: new_y).delete
```

Comment:
Also, I noticed we have a `captured` field.

Instead of deleting the piece, do you want to change it to `captured: true` ?